### PR TITLE
gltf: Remove obsolete hack to embed gltf textures in advanced import

### LINKED ISSUE #75637: Use extracted texture files when saving a mesh's material to a file in Advanced Import Settings
### Describe the project you are working on

Trying to make 3D demo scene with multiple assets

Godot 4 rc2

### Describe the problem or limitation you are having in your project

The current import workflow consumes too much time and storage. 

For GLB/GTLF w/textures the Godot Editor is extracting textures to separate files by default. This is pretty fine - the loading time is fast, reusing textures is available OOTB. But extracting materials creates *.tres files with copies of these textures inside, but saved in text file in super inefficient serialization format  - comma separated decimals in ascii for each byte. This slows down the editor, slows down loading and saving times, and makes editor crashing more often.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

1. Reuse already extracted textures. That would solve described issue.
2. Avoid implicit saving binary files into tres files for any extracted/imported resource.
3. Show a warning message for big files (using a threshold configured in the editor settings, default set to 10kb). 

Just let's assume that storing binary data in text files is wrong by design, and allow doing that only for special cases like storing PackedArrays of hand crafted meshes in the editor.

The above suggestions will solve other similar issues, like storing complex meshes in text files.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

1. Already exported resources are well known for the editor. It should just reference them if they exist instead of serializing same assets into tres file.
2. If binary resources do not exist as a separate files, extract them first.
3. Warn user if the editor is going to store "big" binary data into text file.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No. This is core of the editor and is always used.

Note: the https://github.com/godotengine/godot-proposals/issues/3443 proposal is handy but will not solve this one, because I'm talking here about default behavior.

### Is there a reason why this should be core and not an add-on in the asset library?

As above.

### PR DESCRIPTION
**Pull request by lyuma**
Fixes godotengine/godot#75637 (this should have been a bug, not a feature proposal IMHO)

The override to force embedding of textures within the advanced importer was done as an optimization in a build before caching of extracted textures was implemented: the lack of caching led textures to be re-extracted and/or re-compressed each time. Now that the caching is fixed (for extract textures case), it may be less important.

I believe this fix is acceptable for 4.0 backport: it should have no effect on behavior for existing models or materials. My assumption was that the options passed to advanced import only affected the visual appearance of the import dialog. As it turns out, these settings also affect what is written by the GUI code using `ResourceSaver::save()` when doing material/mesh/animation extraction.

This extraction is only performed by manual user interaction and will not affect existing projects or assets, unless users delete the extracted materials and then extract them again.

"Embed as Basis Universal" may perform poorly in the advanced importer as a result of this change, because basis compression is *not* currently cached between imports, afaik. My opinion is Embed as Basis is not recommended in combination with "Extract Materials" because it will be needlessly embedded in every material.
We could address this performance hit by bringing the override back for the dialog, but we will then have to ban both mesh and material extraction if "Embed as Basis Universal" is used if we do this. I would prefer to keep poor performance and we can perhaps add caching of basis compression later?

### CODE REVIEW

**Review by fire — APPROVED** [review]
The assumption has changed with the advanced importer. It is also extracting materials incorrect in the status quo. This fixes it.

Lyuma will test more.

**Comment by lyuma** [comment]
In testing, I ran into a potential problem in one of my older test projects:
I had a project from 4.0beta which had Extract Textures set (default) but didn't have the textures extracted.

In this case, because the model was already considered to be imported, it had no opportunity to extract textures until I opened the advanced importer. This causes a bug, since textures are extracted but cannot be referenced in time for the advanced import dialog to open:

Here is the specific error log that happens when loading @fire's magic_girl.fbx test scene.
```
WARNING: glTF: Image index '17' couldn't be loaded with the name: nekopin_t. Skipping it.
     at: GLTFDocument::_parse_images (modules\gltf\gltf_document.cpp:3276)
ERROR: Can only append files to import during a current reimport process.
   at: (editor\editor_file_system.cpp:2330)
ERROR: No loader found for resource: res://magic_girl_fbx/magic_girl-60bb0051c272cdf2d62ba4edb599c31c_t_eye_b.png.
   at: (core\io\resource_loader.cpp:230)
```
Basically, the advanced importer screen is not considered a "current reimport process", so `reimport_append` #72455 fails when the advanced import is being rendered. If such a failure happens, clicking Extract Materials will likely fail.

That said, in common usage in which a successful import previously happened before the Advanced Import screen opens, this change seems to be working as intended so far.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.
